src/library-authoring/add-content/AddContentContainer.tsx

```diff
@@ -63,23 +78,123 @@ const AddContentButton = ({ contentType, onCreateContent } : AddContentButtonPro
   );
 };
 
+const AddContentView = ({
+  contentTypes,
+  onCreateContent,
+  isAddLibraryContentModalOpen,
+  closeAddLibraryContentModal,
+}: AddContentViewProps) => {
+  const intl = useIntl();
+  const {
+    collectionId,
+    componentPicker,
+  } = useLibraryContext();
+
+  const collectionButtonData = {
+    name: intl.formatMessage(messages.collectionButton),
+    disabled: false,
+    icon: BookOpen,
+    blockType: 'collection',
+  };
+
+  const libraryContentButtonData = {
+    name: intl.formatMessage(messages.libraryContentButton),
+    disabled: false,
+    icon: Folder,
+    blockType: 'libraryContent',
+  };
+
+  return (
+    <>
+      {collectionId ? (
+        componentPicker && (
+          <>
+            <AddContentButton contentType={libraryContentButtonData} onCreateContent={onCreateContent} />
+            <PickLibraryContentModal
+              isOpen={isAddLibraryContentModalOpen}
+              onClose={closeAddLibraryContentModal}
+            />
+          </>
+        )
+      ) : (
+        <AddContentButton contentType={collectionButtonData} onCreateContent={onCreateContent} />
+      )}
+      <hr className="w-100 bg-gray-500" />
+      {/* Note: for MVP we are hiding the unuspported types, not just disabling them. */}
+      {contentTypes.filter(ct => !ct.disabled).map((contentType) => (
+        <AddContentButton
+          key={`add-content-${contentType.blockType}`}
+          contentType={contentType}
+          onCreateContent={onCreateContent}
+        />
+      ))}
+    </>
+  );
+};
+
+const AddAdvancedContentView = ({
+  closeAdvancedList,
+  onCreateContent,
+  isBlockTypeEnabled,
+  isBasicBlock,
+}: AddAdvancedContentViewProps) => {
+  const intl = useIntl();
+  // We use block types data from backend to verify if the blocks
+  // in `LIBRARY_ADVANCED_BLOCKS` exists. This is to avoid show a non-existent
+  // block in the advanced blocks list.
+  // Also, we use that data to get the translated display name of the block.
+  const { blockTypesData } = useLibraryContext();
```

Comment:
Why are you storing `blockTypesData` in the library context? I think it's more efficient if you just fetch it here using `useBlockTypesMetadata` directly. Otherwise, you may be fetching it when it's not needed; it will be loaded for every time the library UI is opened, but we only need it if the user clicks on "Add Component > Advanced".

Reply:
Updated: https://github.com/openedx/frontend-app-authoring/pull/1653/commits/ab97c029806291f003668ec5bfcb72e403188f1c